### PR TITLE
Plugin system

### PR DESCRIPTION
**Pull request by theoriginalbit**
Wew! Added plugin support and converted Blackjack and Slots over to a plugin.

Plugins will be more robust and stable as there is more error handling.

### CODE REVIEW

File: lib/plugin/core/manager.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      Log.error "#{e.message}"
+    end
+
+  end
+
+  class Plugin
+
+    attr_accessor :name
+
+    # should not be called directly, use PlugMan.define instead.
```
Comment (stephwestgarth) [inline lib/plugin/core/manager.rb]: PlugMan.define is not defined anywhere in this project. Make this comment a bit more meaningful please

**Comment by stephwestgarth** [comment]
Looks good. Can we just make sure that the comments match what actually works? Note added to Plugin class.

File: lib/plugin/core/manager.rb (hunk 1 above)
Comment (theoriginalbit) [inline lib/plugin/core/manager.rb]: was the name I'd originally called it. Next commit has this comment updated.

**Comment by theoriginalbit** [comment]
@sbaker7 fixed the comment in the last two commits

**Comment by stephwestgarth** [comment]
Thank you

**Pull request merged**